js/roles.js

```diff
@@ -0,0 +1,492 @@
+var allRoles;
+var employeeList;
+
+function createRoleCard(data) {
+  const parent = document.querySelector('.role-card-container');
+  let roleCard=createNewElement('div',"role-card","flex-container","flex-col")
+  let roleName=createNewElement('div',"role-title-icon","d-flex","jus-content-btw","w-100")
+  let roleTitle=createNewElement('div','role-title')
+  roleTitle.textContent = data.role;
+  let editIcon=createNewElementWithAttr('img',['src',"../assets/icons/edit.svg"],['alt','edit-icon'])
+  editIcon.addEventListener('click',(event)=>{
+    openEditRoleForm(event)
+  })
+  roleName.appendChild(roleTitle);
+  roleName.appendChild(editIcon);
+  let roleDetails=createNewElement('div',"w-100","role-details","flex-container","flex-col")
+  let roleDepartment=createNewElement('div',"role-department","d-flex","jus-content-btw","w-100")
+  let deptIcon=createNewElement('div','dept-icon','d-flex')
+  let deptImg=createNewElementWithAttr('img',['src',"../assets/icons/emp-id.svg"],['alt','department-icon'])
+  const deptLabel = document.createTextNode("Department");
+  deptIcon.appendChild(deptImg);
+  deptIcon.appendChild(deptLabel);
+  let roleDeptName=createNewElement('div','role-dept-name')
+  roleDeptName.textContent = data.dept;
+  roleDepartment.appendChild(deptIcon);
```

Comment:
After creating all the elements, try to maintain a list and add it to the dom using for loop

Reply:
All resolved